### DISCUSSION TITLE
fandol 字体字库缺少生僻字，时常出现人名无法完整显示

### DISCUSSION THREAD

**Discussion by function2-llx** [tuna/thuthesis#669] [Q&A] [answered]
当使用 fandol 字体时（例如在 ubuntu 下 `fontset` 使用 auto 时），无法正常显示许多生僻字。

例如当使用 fandol 的仿宋字体 `FandolFang` 时，作者值为"荀彧"时，作者区域无法显示“荀彧”的“彧”。

![图片](https://user-images.githubusercontent.com/38486514/119433590-1eaf4f80-bd49-11eb-9cc9-738f39eae49d.png)

如果使用 windows 下的仿宋字体 `simfang.ttf` 就没有问题：

![图片](https://user-images.githubusercontent.com/38486514/119433803-8cf41200-bd49-11eb-8f6a-ee63609cf091.png)

一个个人的解决方案：如果使用 overleaf，`fontset` 值设为 `windows` 的话编译会超时；需要自行找到或下载仿宋字体（如 windows 自带的 `simfang.ttf`）上传到 overleaf 项目中，然后把[这一段](https://github.com/tuna/thuthesis/blob/d627dbe162b30f0e89ba4a583dadee704b7c70fa/thuthesis.dtx#L2580-L2583)替换为 `\setCJKfamilyfont{zhfs}{simfang.ttf}`。

考虑到姓名中出现生僻字这一现象并不罕见，是否考虑将现有 fandol 字体替换为其它能显示生僻字的字体？

**Comment by zepinglee** [comment]
FandolFang 只覆盖到GB 2312，其他字体由于版权问题无法作为模板的一部分发布。

**Reply by function2-llx** [reply]
好吧。那或许文档里可以提及此事，并给出一些解决方案？如果可以，我愿意参与这部分工作。

**Reply by zepinglee** [reply, marked as answer]
文档中有“用户在提交终版前应使用 Windows 平台的字体进行编译。”

**Reply by function2-llx** [reply]
好的，我有注意到那句话，不过因为前面有“研究生院建议”，我最初以为是只针对非做本科毕设的同学。

多谢。
